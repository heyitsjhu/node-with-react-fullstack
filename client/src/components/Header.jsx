import React, { Component } from 'react';

// hook up component to the redux store
import { connect } from 'react-redux';

class Header extends Component {

  // methods defined inside this component,
  // but outside of built-in react functions,
  // are classified as helper methods

  renderContent() {
    switch (this.props.auth) {
      // unsure if user is logged in (still pending)
      case null:
        return;
      // certain user is logged out
      case false:
        return (
          <li>
            <a href="/auth/google">Login with Google</a>
          </li>
        );
      // user is logged in
      default:
        return (
          <li>
            <a href="/api/logout">Logout</a>
          </li>
        );
    }
  }


  render() {
    console.log(this.props);
    return(
      <header>
        <nav>
          <div className="nav-wrapper">
            <a href="#" className="left brand-logo">Welcome</a>
            <ul id="nav-mobile" className="right hide-on-med-and-down">

              {this.renderContent()}

            </ul>
          </div>
        </nav>
      </header>
    )
  }
}

// retrieves the entire redux store, allowing us
// to return just the state we're concerned about
// for this component
// check the reducers index.js page to see which properties (states)
// are available
// the property (state) is then accessible through this.props
function mapStateToProps({ auth }) {
  return { auth };
}

// connect this component to redux store
export default connect(mapStateToProps)(Header);