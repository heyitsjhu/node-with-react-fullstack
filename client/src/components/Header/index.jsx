import React, { Component } from 'react';

// hook up component to the redux store
import { connect } from 'react-redux';

// import use of react-router-dom's Link tag
// to render a new react router component
import { Link } from 'react-router-dom';

import Payments from './Payments';

class Header extends Component {
  // methods defined inside this component,
  // but outside of built-in react functions,
  // are classified as helper methods

  renderContent() {
    switch (this.props.auth) {
      // unsure if user is logged in (still pending)
      case null:
        return;
      // certain user is logged out
      case false:
        return (
          <li>
            <a href="/auth/google">Login with Google</a>
          </li>
        );
      // user is logged in
      default:
        return [
          <li key={1}>
            <Payments />
          </li>,
          <li key={2}>
            <a href="/api/logout">Logout</a>
          </li>
        ];
    }
  }

  render() {
    console.log(this.props);
    return (
      <header>
        <nav>
          <div className="nav-wrapper">
            <Link
              to={this.props.auth ? '/surveys' : '/'}
              className="left brand-logo"
            >
              Welcome
            </Link>

            <ul id="nav-mobile" className="right hide-on-med-and-down">
              {this.renderContent()}
            </ul>
          </div>
        </nav>
      </header>
    );
  }
}

// retrieves the entire redux store, allowing us
// to return just the state we're concerned about
// for this component
// check the reducers index.js page to see which properties (states)
// are available
// the property (state) is then accessible through this.props
function mapStateToProps({ auth }) {
  return { auth };
}

// connect this component to redux store
export default connect(mapStateToProps)(Header);
