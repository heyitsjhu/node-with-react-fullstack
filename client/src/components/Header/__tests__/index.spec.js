import React from 'react';
import Header from '../index';
import * as renderer from 'react-test-renderer';

describe('<Header />', () => {
  it('renders and matches our snapshot', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
