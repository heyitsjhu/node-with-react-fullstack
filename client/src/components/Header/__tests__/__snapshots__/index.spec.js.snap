// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders and matches our snapshot 1`] = `
<header>
  <nav>
    <div
      className="nav-wrapper"
    >
      <a
        className="left brand-logo"
        href="#"
      >
        Welcome
      </a>
      <ul
        className="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="/auth/google"
          >
            Login with Google
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
